src/lib/comms/sol-coap.c

```diff
@@ -104,6 +105,7 @@ struct resource_observer {
 };
 
 struct pending_reply {
+    struct sol_reentrant reentrant;
```

Comment:
you made it work without requiring it as the first member, so for alignment purposes move this as last member.